### ISSUE TITLE
TypeError on modification of model with array field on JavaScriptCore

### ISSUE THREAD

**Issue by sttts** [louischatriot/nedb#267]
On JavaScriptCore based JS engines nedb is not able to modifiy document models with array fields, resulting in an TypeError exception in function `checkKey` in `model.js`:

```
'undefined' is not a function (evaluating 'k.indexOf('.')')
```

**Comment by jthrilly** [comment]
Any chance of integrating this fix? I believe I'm facing this issue on the latest Safari (9).

**Comment by louischatriot** [comment]
Thanks for raising this issue, the fix is now live in nedb v1.1.3. Can you please confirm this does work for you?
